include/dlaf/matrix/print_gpu.h

```diff
@@ -0,0 +1,37 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#ifdef DLAF_WITH_CUDA
+#pragma once
+
+#include <cuda_runtime.h>
+
+#include "dlaf/matrix/tile.h"
+
+namespace dlaf {
+
+namespace matrix {
+
+/// Print a tile in csv format to standard output
+template <class Format, class T>
```

Comment:
Is this more of a test utility? I.e. does it need to be public in the main dlaf includes? If public the doc should possibly say that this synchronizes forcefully and should not be used in the middle of an algorithm?

Reply:
As the CPU print function, the GPU print is a nice to have in any situation.
For debugging, you might want to print a tile (or a matrix (implementation will follow the first time it will be needed)) in the user application as well.
Regarding the synchronization is the same as other functions of tiles:
- if called when scheduling a `.get()` on the future is also needed, and if used with a "clean" stream will not introduce any further synchronization.
- if called in a task, the only difference is that the CPU doesn't only launch the kernels but it does also some work.
  (It can be considered an hybrid task).
  IIRC the cuda tasks are executed from the main pool, therefore even if tasks do not return immediately it shouldn't be a problem.
Anyway as any output operation it introduces some overhead and this should be the main reason to limit its usage.

Reply:
Ok, sounds good. It's just the `cudaStreamSynchronize` that I'm "concerned" about. I don't think you need to do anything about it, but it will block the HPX worker thread (potentially for a long time). I think it's fine for the use cases of this function, but that's why I was wondering if the docs need to mention this behaviour.

Reply:
Thanks the update!

Reply:
Sorry, I was still updating it, now it should be done.

Reply:
My review still stands as approved by the way ;) I don't know if others still have comments though.